### DISCUSSION TITLE
What does cubeviz.specviz.get_spectra() do when multiple spatial/spectra Subsets exist?

### DISCUSSION THREAD

**Discussion by pllim** [spacetelescope/jdaviz#1582] [Q&A] [answered]
![Screenshot 2022-08-17 182316](https://user-images.githubusercontent.com/2090236/185254176-47ac2928-2bf1-4216-9b8e-8772e57a5e48.jpg)

Let's say I have 3 spatial Subsets and then 1 spectral Subset. I see on the screen as above. What should I expect when I run `cubeviz.specviz.get_spectra()`? Right now, I see a dictionary with these keys: `contents[FLUX]`, `Subset 1`, `Subset 2`, `Subset 3`, `Subset 4`. Each one only returns one `Spectrum1D` object each.

| key | type | data | spectrum expectation? |
| --- | --- | --- | --- |
| `contents[FLUX]` | input data | full cube | collapsed cube, no mask |
| `Subset 1` | Spatial | cube covered by spatial region | collapsed cube at spatial region, but is the spectral region mask applied? |
| `Subset 2` | Spectral | cube covered by the spectral region | collapsed cube with wavelength not covered masked out, but is this always the full cube and spatial is ignored? |
| `Subset 3` | Spatial | See `Subset 1` |
| `Subset 4` | Spatial | See `Subset 1` |

Is there a way to grab the spectra that is highlighted by the green lines?

**Comment by kecnry** [comment, marked as answer]
Not directly from a user-friendly API, no, although that is planned for future work (#1490 [:cat:](https://jira.stsci.edu/browse/JDAT-2594)).  To do it manually, you could get the mask from subset 2 and apply it to the fully collapsed spectrum of subset 3 (that's how the highlighting is done under-the-hood by #1528).  The "mask" can be accessed either from the subset object, or by checking `isnan` on the fluxes of the retrieved Subset 2 (in your case above).

**Reply by pllim** [reply]
Okay, thanks for the clarification!
